### PR TITLE
Password Generation

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/0/0/1201969773101195/f
Tech Design URL:
CC:

**Description**:

This PR updates BSK to automatically save generated credentials to the Autofill store when the user selects them.

**Steps to test this PR**:
1. None of the changes in this PR are used by the iOS codebase yet; simply test that email autofill works as expected

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPad

**OS Testing**:

* [ ] iOS 13
* [ ] iOS 14
* [ ] iOS 15

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (THISISDINOSAUR) [inline DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: What happened here?

Comment (samsymons) [reply DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: This is regrettably Xcode working normally, it adds and removes this block on a whim. It's harmless, but I can forcibly reset this change if you want.

Comment (THISISDINOSAUR) [reply DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Since we're not pinned to a specific version, could this backfire (e.g. we without realising end up upping to 1.0.4 and that breaks something)

Comment (samsymons) [reply DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Fortunately we'll be good here before we merge:

* BSK points to an exact version of TrackerRadarKit [here](https://github.com/duckduckgo/BrowserServicesKit/blob/main/Package.swift#L18)
* We do point to an exact version of BSK in the macOS app, just that this branch is pointing to a BSK branch for testing - before this is merged though, this will be updated to point to an exact BSK version

**Review by THISISDINOSAUR — APPROVED** [review]
I tested this as part of reviewing https://github.com/duckduckgo/BrowserServicesKit/pull/81 
so it's good to go once it's pointing at the right BSK version

**Pull request merged**
